### ISSUE TITLE
networkctl reload doesn't reload drop-in files unless the main file has a changed mtime

### ISSUE THREAD

**Issue by Hello71** [systemd/systemd#15521]
with the main config file in /etc/systemd/network, and a drop-in in /run/systemd/network/wg0.network.d, the changes in the drop-in are not applied unless `touch /etc/systemd/network/wg0.network` is run before `networkctl reload`. even if `networkctl delete wg0` is run, the old config file is still used during reload. strangely, both the original and drop-in files are re-read, but the changes are not applied.

**Comment by petrkutalek** [comment]
I can confirm that. 

I have simple configuration for a LTE modem in my laptop. Default configuration `/etc/systemd/network/50-wwp0s20f0u7.network` just ignores that network interface:
```
[Match]
Name=wwp0s20f0u7

[Link]
Unmanaged=true

[Network]
Description=LTE (unconnected)
```
Now I have a script that establishes a LTE connection. It places online configuration into drop-in folder, i.e. `/run/systemd/network/50-wwp0s20f0u7.network.d/online.conf` and overrides basic configuration:
```
[Match]
Name=wwp0s20f0u7

[Link]
Unmanaged=false

[Network]
Description=LTE
Address=_ip_
LinkLocalAddressing=false
LLMNR=false
LLDP=false
DNS=_dns1_
DNS=_dns2_
DNSOverTLS=false

[Route]
Gateway=_gw_
MTUBytes=1500
Metric=250
```
and calls `networkctl reload`. It does not work. Neither if I use `/etc/systemd/network/50-wwp0s20f0u7.network.d/online.conf` although specification says "In addition to /etc/systemd/network, drop-in ".d" directories can be placed in /usr/lib/systemd/network or /run/systemd/network directories. Drop-in files in /etc take precedence over those in /run which in turn take precedence over those in /usr/lib. _Drop-in files under any of these directories take precedence over the main network file wherever located._"
If my script just overwrites original configuration or modifies it then `networkctl` works OK.

**Comment by yuwata** [comment]
Fix is waiting in #16030. If possible, please test the PR. Thank you.
